python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,71 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
+    """
+    !Select sources that are useful for matching.
+
+    Good matching sources have high signal/noise, are non-blended. They need not
+    be PSF sources, just have reliable centroids.
```

Comment:
Will do on the rewrite.

The plan currently is to replace the old matcher with the new matcherPessimsiticB after I run a large amount of data through the new matcher and confirm its performance. Hence why I don't not investigate the effect of the additional cuts on the old matcher. matcherPessimisticSourceSelector is essentially the combination of isGood and isUsable from the old SourceInfo class in matchOptimisticB. matcherSourceSelector is the isUsable part of the SourceInfo class.